util/ranger/types.go

```diff
@@ -216,6 +225,22 @@ func (ran *Range) PrefixEqualLen(sc *stmtctx.StatementContext) (int, error) {
 	return len(ran.LowVal), nil
 }
 
+// EmptyRangeSize is the size of empty range.
+const EmptyRangeSize = int64(unsafe.Sizeof(Range{}))
+
+// MemUsage gets the memory usage of range.
+func (ran *Range) MemUsage() (sum int64) {
+	sum = EmptyRangeSize + int64(cap(ran.LowVal))*types.EmptyDatumSize + int64(cap(ran.HighVal))*types.EmptyDatumSize + int64(cap(ran.Collators))*16
+	for _, val := range ran.LowVal {
+		sum += val.MemUsage() - types.EmptyDatumSize
+	}
+	for _, val := range ran.HighVal {
+		sum += val.MemUsage() - types.EmptyDatumSize
+	}
+	// We ignore size of collator currently.
+	return sum
```

Comment:
If we ignore the size of collator, why not directly like follwing:

```suggestion
        sum = 0
	for _, val := range ran.LowVal {
		sum += val.MemUsage()
	}
	for _, val := range ran.HighVal {
		sum += val.MemUsage()
	}
```

Reply:
Though we ignore the size of Collator, we track the size of the slice `Range.Collators []collate.Collator`.